source/appModules/nvda.py

```diff
@@ -32,14 +32,29 @@ def _get_presentationType(self):
 		return presType
 
 class NvdaSettingsCategoryPanel(IAccessible):
+	# The configuration profile that has been previously edited.
+	# This ought to be a class property.
+	oldProfile = None
 
 	def event_nameChange(self):
 		if self in api.getFocusAncestors():
 			speech.speakObjectProperties(self, name=True, reason=controlTypes.REASON_CHANGE)
 		braille.handler.handleUpdate(self)
+		self._handlePossibleProfileSwitch()
+
+	@classmethod
+	def _handlePossibleProfileSwitch(cls):
+		from gui.settingsDialogs import NvdaSettingsDialogActiveConfigProfile as newProfile
+		if (cls.oldProfile and newProfile and newProfile != cls.oldProfile:

```

Comment:
There is a syntax error here, a missing ')'. This is fixed in https://github.com/BabbageCom/nvda/pull/5

Reply:
Wow, interesting. I must have screwed that up accidentally after I tested it.